packages/desktop-gui/src/lib/routing.jsx

```diff
@@ -1,13 +1,16 @@
 import React from 'react'
+import Promise from 'bluebird'
 
 import viewStore from './view-store'
 
 const Link = ({ children, to, onClick }) => {
   const navigate = (e) => {
     e.preventDefault()
-    if (onClick) onClick()
 
-    to.navigate()
+    Promise.try(() => onClick && onClick())
```

Comment:
Can we instead leverage `defaultProps` and set `onClick` to be a `noop` function/equivalent? This eliminates the need to do these conditional checks all throughout the codebase. We still want it in a try/catch I imagine.